keel-api/src/main/kotlin/com/netflix/spinnaker/keel/api/constraints/ConstraintEvaluator.kt

```diff
@@ -0,0 +1,76 @@
+/*
+ * Copyright 2020 Netflix, Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package com.netflix.spinnaker.keel.api.constraints
+
+import com.netflix.spinnaker.keel.api.Constraint
+import com.netflix.spinnaker.keel.api.DeliveryConfig
+import com.netflix.spinnaker.keel.api.Environment
+import com.netflix.spinnaker.keel.api.artifacts.DeliveryArtifact
+import com.netflix.spinnaker.keel.api.support.EventPublisher
+import com.netflix.spinnaker.kork.plugins.api.internal.SpinnakerExtensionPoint
+
+/**
+ * TODO: Docs
+ */
+interface ConstraintEvaluator<T : Constraint> : SpinnakerExtensionPoint {
+
+  companion object {
+    /**
+     * TODO: Docs
+     */
+    fun <T> getConstraintForEnvironment(
+      deliveryConfig: DeliveryConfig,
+      targetEnvironment: String,
+      klass: Class<T>
```

Comment:
Could you please take this opportunity to rename this to `constraintType` for clarity?